From f5ac5a270c72a9965dad4b5b5422f3d99be9466d Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Thu, 3 Mar 2022 15:32:10 -0800
Subject: [PATCH 2/3] Add ICU build option

The provided CMake doesn't have a build option for ICU so add it.
---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d0b0d15d..c821c780 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,6 +40,7 @@ endif()
 option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 option(LIBXSLT_WITH_DEBUGGER "Add the debugging support" ON)
 option(LIBXSLT_WITH_CRYPTO "Add crypto support to exslt" OFF)
+option(LIBXSLT_WITH_ICU "Add ICU support to xslt" OFF)
 option(LIBXSLT_WITH_MEM_DEBUG "Add the memory debugging module" OFF)
 option(LIBXSLT_WITH_MODULES "Add the module support" ON)
 option(LIBXSLT_WITH_PROFILER "Add the profiling support" ON)
@@ -57,6 +58,10 @@ if(LIBXSLT_WITH_CRYPTO AND NOT WIN32)
 	find_package(Gcrypt REQUIRED)
 endif()
 
+if(LIBXSLT_WITH_ICU)
+	find_package(ICU REQUIRED COMPONENTS data i18n uc)
+endif()
+
 if(LIBXSLT_WITH_PYTHON)
 	check_include_files(unistd.h HAVE_UNISTD_H)
 	check_symbol_exists(F_GETFL fcntl.h HAVE_F_GETFL)
@@ -230,6 +235,9 @@ target_include_directories(
 )
 
 target_link_libraries(LibXslt PUBLIC LibXml2::LibXml2)
+if (LIBXSLT_WITH_ICU)
+	target_link_libraries(LibXslt PUBLIC ICU::uc)
+endif()
 
 if(UNIX)
 	target_link_libraries(LibXslt PRIVATE m)
-- 
2.40.1.windows.1

